index.rst

```diff
@@ -42,20 +42,163 @@
+It will also publish any telemetry from the pipelines themselves.
+
+The OCPS itself acts as a small interface between the OCS and a pipeline execution environment.
+The Observing Script does not have to know how pipelines are executed, all it needs to know is that named pipelines can be executed and will report when they have completed.
+
+In the interim before Gen 3 is fully deployed, the OCPS could be something that launches a command line task.
+It could also be a full Gen 3 pipeline executed by ``pipetask``.
+It is though important that the OCPS knows when processing completes and also knows whether processing succeeded.
+There also needs to be a mechanism for funneling metrics from the pipelines to the OCPS so that they can be issued as telemetry.
+
+Open Questions
+--------------
+
+There are some details that need to be considered:
+
+* If there is a delay in ingesting the previous observation into the OODS, how does the observing script know that the event that was received is for the correct observation?
+  Should it listen for the camera event indicating that the data were taken and read the obsid from that event?
+* If the OCPS is still handling the previous processing request when the new command arrives what should happen?
+  Should it return immediately with a busy status?
+  Should it queue the request?
+  Should it block until the prior processing has completed but timeout if it is taking too long?
+* Should the OCPS check that the data really are in the OODS before accepting the command?
+  There is nothing in the system to prevent an observing script from sending the process command before ingestion.
+* Should the observing script wait for the event from the OCPS indicating that the data were processed to completion before taking more data?
+  Generally it should not wait unless it is interested in the telemetry.
+* If data fail processing for some reason should the observing script be expected to know about it?
+  Should the completion event indicate whether the pipeline completed without error?
+* If data are processed but some QA metric fails, should it be included in the co-add?
+  Failing QA will result in the output data being stored in the Butler repository and would likely result in good pipeline exit status.
+* Should metrics be published to the SQuaSH system in addition to being published as telemetry?
+  Given the transient nature of the summit data products and the presence of the metrics in the EFD via telemetry, it may not be necessary to also store them in SQuaSH.
+* Does SAL require that all metrics that could possibly be reported from a pipeline be declared in advance?
+  Or can metrics be issued as a JSON document and allow any content?
+  Does LOVE work in such a dynamic environment?
+
+Camera Diagnostic Cluster
+=========================
+
+The diagnostic cluster is currently designed to be used by the camera system for their own analysis.
```

Comment:
We really need to sort this out.  My understanding is that pre-processing JPEGs is the only use-case for the diagnostic camera and that they'll be using LSST-compliant processing on the mountain.  I certainly fervently hope that this is true.